packages/grafana-data/src/themes/createColors.ts

```diff
@@ -0,0 +1,200 @@
+import { merge } from 'lodash';
+import { palette } from './palette';
+import { DeepPartial } from './types';
+
+export interface ThemeRichColor {
+  /** color intent (primary, secondary, info, error, etc) */
+  name: string;
+  /** Main color */
+  main: string;
+  /** Light shade */
+  light: string;
+  /** Dark shade */
+  dark: string;
```

Comment:
No sure I see the point of these two  According to @Ijin08 and @hemdrup work I dont think we have a concept of shades of a color, right? 

I would start slim with this interface and keep it minimal with `name` and `main` properties to start with.

Reply:
Yes, but to derive their button and alert variables we need more here , for example a border and text color based on the main color

Reply:
https://www.figma.com/file/XoSPNXemxDq0aSNt0ersM2/New-theme-system?node-id=90%3A4249

almost every color have 2 or 3 shades (including the main)

Reply:
Then it might be my misunderstanding of the intent of the reach color. I was assuming this model represents the set of tokens we want to establish. So, to be 1-1 with the proposal from @Ijin08 there is no such thing like primary[dark|light] shade, as long as we do not think of dark/light theme differentiation. 

Given https://www.figma.com/file/XoSPNXemxDq0aSNt0ersM2/New-theme-system?node-id=258%3A11125 I believe the reach color can be used for the "palette" while the $primary/$secondary etc would use the palette defined colors.

Reply:
I would imagine the model flow to be the following:

1. Palette - `typeof ThemeRichColor` - it makes sense to include "shades" of given palette color
2. Colors (i.e. primary, secondary, success, etc) -  `typeof string`, derived from the "palette". Representing what we call internaly as  **tokens**.

I also like the idea of containing the contrastText within the ThemeRichColor, but I'm afraid this will bring some complication and probably is a source of some duplication. Thinking in particular of a case where the palette becomes dependant on the text color tokens, what breaks the flow I was thinking about above. 

The case I have in mind:
```
const green: ThemeRichColor = {
  main: #00ff00;
  dark: lighten(#00ff00, X);
  light: darken(#00ff00, X);
  text: ??? // do we refer to a TOKEN here, or do we use a static rgb(a)?
}
```

Reply:
I this was attempt to explore a unified set of high level colors , needs more thought and exploration to see if this can fulfill the same roles as the button and state colors (see my own comment on this higher up in the review)

Reply:
@dprokop 
<img width="1131" alt="Screen Shot 2021-03-26 at 12 52 23" src="https://user-images.githubusercontent.com/10999/112627638-28443600-8e32-11eb-99b7-86625f47aa81.png">

Reply:
yup yup, saw that :) Just adding my two cents to the thought process for this :)

Reply:
> I also like the idea of containing the contrastText within the ThemeRichColor, but I'm afraid this will bring some complication and probably is a source of some duplication.

Why? otherwise, we will need a separate variable for it (or call the function getTextForBg). contrastText will be used for button text colors

Reply:
Was thinking about this particular example:

```
const green: ThemeRichColor = {
  main: #00ff00;
  dark: lighten(#00ff00, X);
  light: darken(#00ff00, X);
  text: ??? // do we refer to a TOKEN here, or do we use a static rgb(a)?
}
```

And when I say TOKEN i mean $textXXX token. But I assume the $textXXX tokens can be independent from the rich color's `text` property and be derived directly from the palete, while as you are suggesting, the `text` property would be used in scenarios when the contrast between the text and the background needs to be ensured.

Reply:
> Palette - typeof ThemeRichColor - it makes sense to include "shades" of given palette color
> Colors (i.e. primary, secondary, success, etc) - typeof string, derived from the "palette". Representing what we call internaly as tokens.

Yea, trying to keep the organization we have to not change too much. Palette => the raw string colors as just named token gray10, gray15, red88 etc.

Example 
theme.colors.primary.main  -> used for primary button background
theme.colors.primary.contrastText -> used for primary button text color 
theme.colors.primary.text  (a hypothetic variant of main used for text) -> used for link colors 
theme.colors.primary.border   (a hypothetic variant of main used for borders, maybe same as text)

not sure about any of this just exploring ways we can avoid btnPrimaryBackground, btnPrimaryText, btnDestructiveBackground, etc, and just take those from these rich colors

Reply:
But this is all a very early test to play around with the code architecture & typescript typings to support a rich color object what allow the user (and us) to partially supply colors and the rest can be calculated if left empty (main is required). Next is to setup a storybook to play around with this with some fake components.

Reply:
> And when I say TOKEN i mean $textXXX token. But I assume the $textXXX tokens can be independent from the rich color's text property and be derived directly from the palete, while as you are suggesting, the text property would be used in scenarios when the contrast between the text and the background needs to be ensured.

not sure I understand. the contrastText is basically either white or dark depending on how light or dark the main color is. We will measure the luminance of the main color to derive the contrastText (and use the primary color from dark or light theme as way to pick white or light )